aten/src/ATen/native/mkldnn/Conv.cpp

```diff
@@ -66,28 +63,16 @@ ideep::tensor _mkldnn_conv2d(
     at::IntArrayRef stride,
     at::IntArrayRef dilation,
     int64_t groups) {
-  std::vector<int64_t> kernel_size(x.ndims());
-  // mkldnn conv2d weights could have been re-ordered to 5d by
-  // mkldnn_reorder_conv2d_weight
-  if (w.ndims() == x.ndims() + 1) {
```

Comment:
did we verify that we never serialized 5d weights previously using JIT serialization? It seems ok (because we do custom setstate in mkldnn_utils.py), but I'd double check that JIT serialized models with grouped convs from previous pytorch can be loaded with this PR

Reply:
We **do** have serialized models with 5d weights (e.g. ResNeXt101 which has grouped conv) and https://github.com/fairinternal/ClassyVision uses it.

Reply:
> We do have serialized models with 5d weights

@dzhulgakov @bddppq: oh, this doesn't sound right. The custom [serialization code](https://github.com/pytorch/pytorch/blob/master/torch/utils/mkldnn.py#L58) does the right thing to save dense tensors but `to_dense` should return a CPU tensor with compatible format reusable by other devices, e.g. 4-d weight instead of 5-d weight. Shall we fix it this time?

Reply:
yes, it wasn't right and got fixed. Unfortunately we still have old serialized models.

can you check that the previous serialized models work? Because we do serialize the weight at 5d - see example https://colab.research.google.com/drive/1Fb0I6vmx-0eXZm507yiR0IN_WJ4BYDaz

However, the only place that weight goes into is mkldnn_reorder_conv2d_weight which apparently used to accept 5d input. We might need to add a patch there. Can you add a unittest verifying that the previous 5d serialized input is handled correctly? Thanks!

Reply:
Thanks for the repro. It's very enlightening.

We didn't take that `RecursiveScriptModule` into account indeed, and we noticed that the recursive module `second.pt` cannot be loaded again, as `mkldnn_reorder_conv2d_weight` actually only accept 4d tensors before. Kind of a potential bug though.

We fixed this by safeguarding the `mkldnn_reorder_conv2d_weight` where we load conv weights. It can accept both 4d and 5d weights now, and all conv2d weights will all have a 4d size, so we can further clean up some workaround in `Convolution.cpp`.

@dzhulgakov @bddppq Please take a look at the following commit about this issue.
```
Commit:
dnnl upgrade: support legacy group conv2d
``